### PR TITLE
Sync IntegrationEnrollKind

### PR DESCRIPTION
**Pull request by bernardjkim**
Requires https://github.com/gravitational/cloud/pull/10599

Metrics for integration enrollment does not seem to be working as expected. When I was testing the Datadog plugin enrollment it was reporting enrollment of the Machine ID Azure enrollment.

This seems to be due to the two `IntegrationEnrollKind` enums not being in sync.

~Anyone know if we need to maintain the two separate enums instead of using just one?~

### CODE REVIEW

**Comment by bernardjkim** [comment]
Hmm, would it be a bad idea to rewrite the enum values?
```
Error: Enum value "18" on enum "IntegrationEnrollKind" changed name from "INTEGRATION_ENROLL_KIND_SERVICENOW" to "INTEGRATION_ENROLL_KIND_MACHINE_ID_AWS".
Error: Enum value "19" on enum "IntegrationEnrollKind" changed name from "INTEGRATION_ENROLL_KIND_ENTRA_ID" to "INTEGRATION_ENROLL_KIND_MACHINE_ID_GCP".
Error: Enum value "20" on enum "IntegrationEnrollKind" changed name from "INTEGRATION_ENROLL_KIND_DATADOG_INCIDENT_MANAGEMENT" to "INTEGRATION_ENROLL_KIND_MACHINE_ID_AZURE".
Error: buf found 3 breaking changes.
```

**Comment by Joerger** [comment]
> Should we just reserve these values and start from 25?
Yeah just do that.

File: api/proto/teleport/usageevents/v1/usageevents.proto [hunk 1]
```diff
@@ -570,9 +570,14 @@ enum IntegrationEnrollKind {
...
-  INTEGRATION_ENROLL_KIND_ENTRA_ID = 19;
-  INTEGRATION_ENROLL_KIND_DATADOG_INCIDENT_MANAGEMENT = 20;
+  INTEGRATION_ENROLL_KIND_MACHINE_ID_AWS = 18;
+  INTEGRATION_ENROLL_KIND_MACHINE_ID_GCP = 19;
+  INTEGRATION_ENROLL_KIND_MACHINE_ID_AZURE = 20;
+  INTEGRATION_ENROLL_KIND_MACHINE_ID_SPACELIFT = 21;
+  INTEGRATION_ENROLL_KIND_MACHINE_ID_KUBERNETES = 22;
+  INTEGRATION_ENROLL_KIND_ENTRA_ID = 23;
+  INTEGRATION_ENROLL_KIND_DATADOG_INCIDENT_MANAGEMENT = 24;
+  INTEGRATION_ENROLL_KIND_SERVICENOW = 25;
```
Comment (Joerger) [inline api/proto/teleport/usageevents/v1/usageevents.proto:580]: I think we should just accept that these new Machine ID enums are separated from the older ones, not a big deal really.

```suggestion
  INTEGRATION_ENROLL_KIND_SERVICENOW = 18;
  INTEGRATION_ENROLL_KIND_ENTRA_ID = 19;
  INTEGRATION_ENROLL_KIND_DATADOG_INCIDENT_MANAGEMENT = 20;
  INTEGRATION_ENROLL_KIND_MACHINE_ID_AWS = 21;
  INTEGRATION_ENROLL_KIND_MACHINE_ID_GCP = 22;
  INTEGRATION_ENROLL_KIND_MACHINE_ID_AZURE = 23;
  INTEGRATION_ENROLL_KIND_MACHINE_ID_SPACELIFT = 24;
  INTEGRATION_ENROLL_KIND_MACHINE_ID_KUBERNETES = 25;
```

Comment (Joerger) [reply api/proto/teleport/usageevents/v1/usageevents.proto:580]: Hmm I see now that the issue is that we're out of sync with prehog proto... I'm not sure off the top of my head what the right thing to do here is.

**Comment by zmb3** [comment]
@strideynet any idea what happened here?

It looks like you added some new enum variants to the cloud proto in https://github.com/gravitational/cloud/pull/6792/ with a note that _the RPC handler converts this enum to a string representation, so no changes are actually needed to the handling code_, but the variants were never added on the Teleport side. Was this intentional?

**Review by Joerger — APPROVED** [review]

**Comment by strideynet** [comment]
> Edit: I see they were added to Teleport in https://github.com/gravitational/teleport/pull/34985 but only in the teleport.proto and not in the usageevents.proto. What a footgun we have here.

Ah yeah wow - that'll be the root of the issue here. This seems like a mistake that's a little too easy to mistake - is there a reason that one of these protos can't just reference the other one as a dependency ?

**Review by marcoandredinis — APPROVED** [review]

**Review by kimlisa — APPROVED** [review]

**Comment by bernardjkim** [comment]
> is there a reason that one of these protos can't just reference the other one as a dependency ?

Sorry, I let this PR get stale. I'll go ahead and merge this for now to fix metrics collection. We can revisit this topic in a separate PR.

**Pull request merged**